cmd/state-exec/comm.go

```diff
@@ -13,23 +13,23 @@ const (
 )
 
 func sendMsgToService(sockPath string, hb *svcmsg.Heartbeat) error {
-	ef := "send msg to service: %w"
+	efmt := "send msg to service: %w"
```

Comment:
Please write custom errors for each. The point of this is to allow us to trace the error, if multiple share the same error then that becomes more difficult.

Recall we used to include stacktraces. We removed those when you argued that those were unnecessary if we provide wrapped errors from which we can extrapolate the stack.

Reply:
Darn. Heh. I was hoping this would fly under the radar a bit. My short reply is that the error messages produced by this approach are very clear and contain the entire stack for digging into issues.

Longer reply, if you care for a preview of a Weekly topic: During Innovation Days I worked on communicating why this approach provides improved output, and is easier to use in the code. Two docs have been posted in the Weekly agenda that cover "confessing" versus "blaming" when annotating errors. The first is a visualization of the "alignment of responsibilities", and the other is code examples.

Reply:
We descoped that from this PR though, meaning this PR still has to follow our best practices and not make assumptions about the future or the underlying errors.